src/domain-functions.ts

```diff
@@ -130,5 +130,27 @@ const map: Map = (dfn, mapper) => {
     }
   }
 }
+type MapError = <O>(
+  dfn: DomainFunction<O>,
+  mapper: (element: ErrorData) => ErrorData,
```

Comment:
Looking at the implementation I don't understand why the tsc didn't complain about this:
```suggestion
  mapper: (element: ErrorResult) => ErrorData,
```

Reply:
`ErrorData` fits in `ErrorResult`. For the mapper function `ErrorData` makes more sense since we don't wan't the developer to receive or return the `success` boolean.